### PR TITLE
Load indicator

### PR DESCRIPTION
**Pull request by xwatkins**
### Reference to existing issue
Add a loading and no results indicator

### Description of changes
Added spinner svg (from franklin), and styles.
Use flags to decide what to display and when.
Added examples with edge cases in the `index.html`

### Testing/Styleguide
 - [x] Tests pass
 - [x] Linters pass

### CODE REVIEW

File: src/protvista-uniprot-structure.ts [hunk 1]
```diff
@@ -140,6 +149,15 @@ class ProtvistaUniprotStructure extends LitElement {
...
 
+  addStyles() {
+    // We are not using static get styles()
+    // as we are not using the shadowDOM
+    // because of Mol*
+    const styleTag = document.createElement('style');
+    styleTag.innerHTML = loaderStyles.toString();
+    document.querySelector('head')?.append(styleTag);
+  }
+
```
Comment (aurel-l) [inline src/protvista-uniprot-structure.ts:160]: This should probably cleaned up afterwards if added in the global scope (using connectedCallback and disconnectedCallback).
Either that, or be added once on module load

Comment (aurel-l) [reply src/protvista-uniprot-structure.ts:160]: Also, if the styles are not gonna be scoped in shadow DOM, the selectors should probably be more specific (`protvista-uniprot-structure .my-class` for example)

File: src/protvista-uniprot-structure.ts [hunk 2]
```diff
@@ -140,6 +149,15 @@ class ProtvistaUniprotStructure extends LitElement {
     protvistaDatatableElt.selectedid = this.pdbId;
   }
 
+  addStyles() {
+    // We are not using static get styles()
+    // as we are not using the shadowDOM
+    // because of Mol*
+    const styleTag = document.createElement('style');
+    styleTag.innerHTML = loaderStyles.toString();
+    document.querySelector('head')?.append(styleTag);
```
Comment (dlrice) [inline src/protvista-uniprot-structure.ts:158]: Could it ever happen where styles are appended multiple times to the `head`?

File: src/protvista-uniprot.ts [hunk 3]
```diff
@@ -228,12 +163,21 @@ class ProtvistaUniprot extends LitElement {
...
-            // TODO handle this better based on error code
-            // Fail silently for now
-            (error) => console.warn(error)
-          )
+          load(url.replace('{accession}', accession))
+            .then(
+              (data) => {
+                this.rawData[url] = data.payload;
+                // Some endpoints return empty arrays, while most fail 🙄
+                if (!this.hasData && data.payload?.features.length)
```
Comment (dlrice) [inline src/protvista-uniprot.ts:171]:
```suggestion
                if (!this.hasData && data.payload?.features?.length)
```
Saw an undefined error with your example:
```
      <!-- No features -->
      <protvista-uniprot accession="A0A2K5ULD0"></protvista-uniprot>
```

File: src/protvista-uniprot-structure.ts (hunk 2 above)
Comment (aurel-l) [reply src/protvista-uniprot-structure.ts:158]: Related to https://github.com/ebi-webcomponents/protvista-uniprot/pull/35#discussion_r660408723

File: src/protvista-uniprot-structure.ts (hunk 1 above)
Comment (xwatkins) [reply src/protvista-uniprot-structure.ts]: Added logic to only add styles once and remove when disconnected. I'll make a Jira about the specificity of classes - as that was not changed.
